### PR TITLE
Add tests for the guide book using `docmatic`

### PR DESCRIPTION
**Pull request by althonos**
Related to #84,

I used `docmatic` to generate tests for the documentation, and updated the failing test cases to work with modern versions of the library. Some tests are ignored because they won't compile (for instance the example for `Into<PyErr>` implementation will not compile since the `std::io::Error` was not declared in our module).

### CODE REVIEW

**Comment by fafhrd91** [comment]
could you add config to the travis config, so travis runs tests on each commit?

**Comment by althonos** [comment]
@fafhrd91 : once this is merged all tests will be invoked through `cargo test`, nothing needs to be changed in the CI workflow

**Comment by fafhrd91** [comment]
Good. Thanks

File: guide/src/class.md [hunk 1]
```diff
@@ -72,14 +73,24 @@ To declare constructor, you need to define class method and annotate it with `#[
 attribute. Only python `__new__` method can be specified, `__init__` is not available.
 
 ```rust
+# #![feature(proc_macro, specialization, const_fn)]
+# extern crate pyo3;
+# use pyo3::prelude::*;
+# #[py::class]
```
[the remark this answers is not in the document]
Comment (althonos) [reply guide/src/class.md]: I'll rebase against the current master then, and remove other occurences. Thanks for the hint !

File: Cargo.toml [hunk 2]
```diff
@@ -25,6 +25,9 @@ spin = "0.4.6"
 num-traits = "0.2"
 pyo3cls = { version = "^0.2.1" }
 
+[dev-dependencies]
+docmatic = { version = "^0.1", git = "https://github.com/assert-rs/docmatic" }
```
[the remark this answers is not in the document]
Comment (althonos) [reply Cargo.toml]: I also need to bump the required version to `0.2` since I need features I PR'd into the latest release :wink:

**Comment by althonos** [comment]
Merging before I need to rebase again. Thanks @messense @konstin @fafhrd91 !

**Pull request merged**